package/lib/src/beam_interceptor_pop_scope.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:beamer/beamer.dart';
+import 'package:flutter/widgets.dart';
+
+/// This works like [PopScope], but with beam-interceptors.
+///
+/// See [BeamInterceptor] for more information.
+///
+/// If any of the interceptors return true, the pop will not be invoked.
+///
+/// This works on Navigator.maybePop as well as all the Beamer.beamTo, Beamer.beamToNamed, Beamer.beamBack... operations.
+///
+/// ```dart
+/// BeamInterceptorPopScope(
+///   interceptors: [BeamInterceptor(...), ...],
+///   child: Center(
+///     child: ElevatedButton(
+///       child: const Text('Go back'),
+///       onPressed: () => Beamer.of(context).beamToNamed('/'),
+///     ),
+///   ),
+/// );
+/// ```
+class BeamInterceptorPopScope extends StatefulWidget {
```

Comment:
We should probably calls this just `BeamInterceptorScope` as it's not specific to just `pop`.

Reply:
✅